src/sql/src/plan/statement/ddl.rs

```diff
@@ -3337,6 +3358,24 @@ fn ensure_cluster_is_not_linked(
     }
 }
 
+fn is_storage_cluster(scx: &StatementContext, cluster: &dyn CatalogCluster) -> bool {
+    cluster.bound_objects().iter().all(|id| {
```

Comment:
Why is this not using `any(...)`?

Reply:
We discussed elsewhere, and I put up a fix in #17306.